src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamAppealFeeDueNoRemissionHuEaEuPersonalisation.java

```diff
@@ -0,0 +1,103 @@
+import static uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCaseDefinition.NOTIFICATION_ATTACHMENT_DOCUMENTS;
+
+@Slf4j
+@Service
+public class DetentionEngagementTeamAppealFeeDueNoRemissionHuEaEuPersonalisation implements EmailWithLinkNotificationPersonalisation {
+
+    private final String internalDetainedAppealFeeDueTemplateId;
+    private final DocumentDownloadClient documentDownloadClient;
+    private final DetEmailService detEmailService;
+    private String subjectPrefix;
+
+    public DetentionEngagementTeamAppealFeeDueNoRemissionHuEaEuPersonalisation(
+            @Value("${govnotify.template.pendingPaymentEaHuEu.detentionEngagementTeam.email}") String internalDetainedAppealFeeDueTemplateId,
+            DetEmailService detEmailService,
+            DocumentDownloadClient documentDownloadClient,
+            @Value("${govnotify.emailPrefix.nonAdaInPerson}") String subjectPrefix
+    ) {
+        this.internalDetainedAppealFeeDueTemplateId = internalDetainedAppealFeeDueTemplateId;
+        this.detEmailService = detEmailService;
+        this.documentDownloadClient = documentDownloadClient;
+        this.subjectPrefix = subjectPrefix;
+    }
+
+    @Override
+    public String getTemplateId() {
+        return internalDetainedAppealFeeDueTemplateId;
+    }
+
+    @Override
+    public Set<String> getRecipientsList(AsylumCase asylumCase) {
+        Optional<String> detentionFacility = asylumCase.read(DETENTION_FACILITY, String.class);
+        if (detentionFacility.isEmpty() || detentionFacility.get().equals("other")) {
+            return Collections.emptySet();
+        }
+
+        return Collections.singleton(detEmailService.getDetEmailAddress(asylumCase));
+    }
+
+    @Override
+    public String getReferenceId(Long caseId) {
+        return caseId + "_INTERNAL_DETAINED_APPEAL_FEE_DUE_DET";
+    }
+
+    @Override
+    public Map<String, Object> getPersonalisationForLink(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        return ImmutableMap
+                .<String, Object>builder()
+                .put("subjectPrefix", subjectPrefix)
+                .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("homeOfficeReferenceNumber", asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("appellantGivenNames", asylumCase.read(AsylumCaseDefinition.APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+                .put("appellantFamilyName", asylumCase.read(AsylumCaseDefinition.APPELLANT_FAMILY_NAME, String.class).orElse(""))
+                .put("documentLink", getInternalDetainedAppealFeeDueDocumentInJsonObject(asylumCase))
+                .build();
+    }
+
+    private JSONObject getInternalDetainedAppealFeeDueDocumentInJsonObject(AsylumCase asylumCase) {
```

Comment:
In AsylumUtils we have something similar getLetterForNotification method, could we use it?